google/cloud/aiplatform/base.py

```diff
@@ -94,6 +106,25 @@ def _instantiate_client(
             prediction_client=cls._is_client_prediction_client,
         )
 
+    def _get_gca_resource(self, resource_name):
+        """Returns GAPIC service representation of client class resource."""
```

Comment:
This doesn't necessarily return the resource it sets the `self._gca_resource` attribute.

It may be worth breaking this down into two methods.

`_get_gca_resource` that gets the resource and returns it.

and `_sync_gca_resource` which calls `_get_gca_resource` and sets it to `self._gca_resource`.